include/battle.h

```diff
@@ -709,6 +709,8 @@ STATIC_ASSERT(sizeof(((struct BattleStruct *)0)->palaceFlags) * 8 >= MAX_BATTLER
         typeArg = gBattleMoves[move].type;                            \
 }
 
+#define IS_TYPE_PHYSICAL(moveType)(moveType < TYPE_MYSTERY)
+#define IS_TYPE_SPECIAL(moveType)(moveType > TYPE_MYSTERY)
```

Comment:
Please remove these.

Reply:
There's a slight inconsistency wherever `IS_MOVE_PHYSICAL`/`IS_MOVE_SPECIAL` are used. We're trying to make it so that the hacker can use vanilla Gen 3 mechanics, but I don't see any way to revert the Physical/Special/Status Split. I was wary of adding these because of that.

Reply:
> There's a slight inconsistency wherever IS_MOVE_PHYSICAL/IS_MOVE_SPECIAL are used.

What inconsistency?

> We're trying to make it so that the hacker can use vanilla Gen 3 mechanics, but I don't see any way to revert the Physical/Special/Status Split.

[The expansion has a config to disable the modern PSS Split](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/include/config/battle.h#L64).

Reply:
The reason I thought this was because of `IS_TYPE_PHYSICAL`/`IS_TYPE_SPECIAL` being gone, but I noticed the expansion just checks for if the move type constant is above or below `TYPE_MYSTERY`. I don't see a reason why they should be removed, but _shrug_. It was probably also because in the code in `battle_util.c`, things like the Choice Band use `IS_MOVE_PHYSICAL` and don't check for `B_PHYSICAL_SPECIAL_SPLIT`.

Reply:
> The reason I thought this was because of IS_TYPE_PHYSICAL/IS_TYPE_SPECIAL being gone

You thought there was an inconsistency with the `IS_MOVE_PHYSICAL` and `IS_MOVE_SPECIAL` labels "because of IS_TYPE_PHYSICAL/IS_TYPE_SPECIAL being gone"? I don't understand what do you mean tbh.

> I don't see a reason why they should be removed

They should be removed because they don't have an actual purpose here. Not when you can just use `IS_MOVE_PHYSICAL` or `IS_MOVE_SPECIAL` based on the context.

> It was probably also because in the code in battle_util.c, things like the Choice Band use IS_MOVE_PHYSICAL and don't check for B_PHYSICAL_SPECIAL_SPLIT.

I have no idea what does that have to do with the changes you made in this PR.

Reply:
> I have no idea what does that have to do with the changes you made in this PR.
It's relevant because Light Ball in Gen 3 (i.e. without the PSS) only boosts moves to the right of `TYPE_MYSTERY`, but there's no config for that here, and the same goes for things like Choice Band. I mentioned this in the second commit.

Reply:
> Light Ball in Gen 3 (i.e. without the PSS) only boosts moves to the right of TYPE_MYSTERY

The Light Ball in Gen. 3 boosts Pikachu's Special Attack and has no type-specific checks, therefore, to implement the equivalent effect in the expansion you use `IS_MOVE_SPECIAL` for its check.

[Here's the snippet for the Light Ball's effect in the vanilla codebase](https://github.com/pret/pokeemerald/blob/master/src/pokemon.c#L3190-L3191).